### ISSUE TITLE
Why not pick pure Keras or TF.Keras for ML Research in Google Brain

### ISSUE THREAD

**Issue by innat** [tensorflow/models#9446]
**Google Brain** (GB) publishes state of the art research work, most of the time in pure `TensorFlow`. But unfortunately, the original open-source implementation with pure `TF` is really hard to follow up or unclear or sometimes poorly documented. 

On the other hand, of these honorable work from **GB**, `PyTorch` practitioner feasibly re-implement the idea or published work. Those `PyTorch` implementation (most of the time) is well organized, well-optimized, well documented, and maintained. And even integrate more advanced strategies to enhance the model generalization. 

That brings the main question, why **GB** is not picking the `Keras` as their primary research framework? Is `Keras` not fit enough for ML Research? I believe that `Keras` or `TF.Keras` code would be more clearer than pure `TF`. 

@fchollet  @Wookai  @jhseu

**Comment by saberkun** [comment]
Keras is a high level framework where TF/Pytorch is the general purpose framework.
TF in TF 2.x standardize the model building with tf.Keras (based on tf.Module, similarly sonnet v2). Torch and TF can both be very flexible and code quality is up to the model authors.
Some of the old projects are using old TF style and researchers are not maintaining them.
Please comment to https://github.com/google-research for the brain open source model quality as how to write models is the researchers' decision. 
Thanks

**Comment by innat** [comment]
That's it! 
IMHO, that really doesn't answer the question. 

What do you mean by 
> Please comment to @google-research for the brain open source model quality as how to write models is the researchers' decision.

**Comment by innat** [comment]
@saberkun 
Thanks for your comment. However, I was expecting some detailed answers on this. Your response is quite vague, no offense. 

It would be appreciated if some experienced guy (from **GB** or else) puts some comments on the issue. Specifically, I am looking for details answer on how much pure `TensorFlow` and `Keras` differ from the **GB** research perspective? How much the **GB** teams use `Keras`? If **GB** has got some model idea, why pick pure `TensorFlow` instead of `Keras` to implement? I know in some content `Keras` can be comparatively slower than `TF` (or No, please correct me). In the end, most users use `Keras` application models. Can it be changed to `TensorFlow 2`?

My questions maybe not organized properly, but I hope you get my point.
